src/os/ObjectStore.h

```diff
@@ -766,6 +767,36 @@ class ObjectStore {
     const ghobject_t &oid  ///< [in] object
     ) = 0;
 
+  /**
+   * Iterate object map with user-provided callable
+   *
+   * Warning!  The callable is executed under lock on filestore
+   * operations in c.  Do not use filestore methods on c while
+   * when iterating.  (Filling in a transaction is no problem).
+   *
+   * @return error code, zero on success
+   */
+  struct omap_iter_seek_t {
+    std::string seek_position;
+    enum {
+      LOWER_BOUND,
+      UPPER_BOUND
+    } seek_type = LOWER_BOUND;
+    static omap_iter_seek_t min_lower_bound() { return {}; }
+  };
+  enum class omap_iter_ret_t {
+    STOP,
+    NEXT
+  };
+  virtual int omap_iterate(
+    CollectionHandle &c,   ///< [in] collection
+    const ghobject_t &oid, ///< [in] object
+    omap_iter_seek_t start_from, ///< [in] where the iterator should point to at the beginning
+    std::function<omap_iter_ret_t(std::string_view, std::string_view)> f
+  ) {
```

Comment:
There is something missing in this interface that users of OmapIterator could do.
When user of OmapIterator wanted to retireve exactly 30 elements it could check if there is one more.
If not, it returned in message that "it is all"
```
PrimaryLogPG::do_osd_ops()
...
 case CEPH_OSD_OP_OMAPGETVALS:
 ...
 bool truncated;
 ```

Reply:
I believe `truncated` is already covered: https://github.com/ceph/ceph/pull/60278/files#diff-04d5a3324b706d64a3230221d2eb08f6aa66829f93ca151ebb71896fb743ada7R7776.